consensus/vbft/chain_store.go

```diff
@@ -48,13 +72,50 @@ func (self *ChainStore) GetChainedBlockNum() uint32 {
 	return self.chainedBlockNum
 }
 
+func (self *ChainStore) SetExecMerkleRoot(blkNum uint32, merkleRoot common.Uint256) {
+	if blk, present := self.pendingBlocks[blkNum]; blk != nil && present {
+		blk.execResult.MerkleRoot = merkleRoot
+	} else {
+		log.Errorf("SetExecMerkleRoot failed blkNum:%d,merkleRoot:%s", blkNum, merkleRoot.ToHexString())
+	}
+}
+
+func (self *ChainStore) GetExecMerkleRoot(blkNum uint32) common.Uint256 {
+	if blk, present := self.pendingBlocks[blkNum]; blk != nil && present {
+		return blk.execResult.MerkleRoot
+	}
+	merkleRoot, err := self.server.ledger.GetStateMerkleRoot(blkNum)
```

Comment:
self.db == self.server.ledger

you can get MerkleRoot with self.db.GetStateMerkleRoot

Reply:
ok